x/nodes/keeper/reward.go

```diff
@@ -4,147 +4,48 @@ import (
-		if err != nil {
-			panic(err)
-		}
-		// mint the coins for the dao to the module account
-		err = k.AccountKeeper.MintCoins(ctx, types.ModuleName, daoRewardCoins)
-		if err != nil {
-			panic(err)
-		}
-		// send to validator
-		if err := k.AccountKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, proposerValidator.GetAddress(), propRewardCoins); err != nil {
-			panic(err)
-		}
-		// send to rest dao
-		if err := k.AccountKeeper.SendCoinsFromModuleToModule(ctx, types.ModuleName, govTypes.DAOAccountName, daoRewardCoins); err != nil {
-			panic(err)
-		}
-		logger.Info(fmt.Sprintf("minted %s to block proposer: %s", propRewardCoins.String(), proposerValidator.GetAddress().String()))
-		logger.Info(fmt.Sprintf("minted %s to DAO", daoRewardCoins.String()))
-		ctx.EventManager().EmitEvent(
-			sdk.NewEvent(
-				types.EventTypeProposerReward,
-				sdk.NewAttribute(sdk.AttributeKeyAmount, proposerReward.String()),
-				sdk.NewAttribute(types.AttributeKeyValidator, proposerValidator.GetAddress().String()),
-			),
-		)
-		ctx.EventManager().EmitEvent(
-			sdk.NewEvent(
-				types.EventTypeDAOAllocation,
-				sdk.NewAttribute(sdk.AttributeKeyAmount, daoReward.String()),
-			),
-		)
-	} else {
-		logger.Error(fmt.Sprintf(
-			"WARNING: Attempt to allocate proposer rewards to unknown proposer %s. "+
-				"This should happen only if the proposer unstaked completely within a single block, "+
-				"which generally should not happen except in exceptional circumstances (or fuzz testing). "+
-				"We recommend you investigate immediately.",
-			previousProposer.String()))
-	}
-}
-
-func (k Keeper) GetTotalCustomValidatorAwards(ctx sdk.Ctx) sdk.Int {
-	total := sdk.ZeroInt()
-	store := ctx.KVStore(k.storeKey)
-	iterator := sdk.KVStorePrefixIterator(store, types.AwardValidatorKey)
-	defer iterator.Close()
-	for ; iterator.Valid(); iterator.Next() {
-		amount := sdk.Int{}
-		k.cdc.MustUnmarshalBinaryBare(iterator.Value(), &amount)
-		total = total.Add(amount)
+	feesCollector := k.getFeePool(ctx)
+	feesCollected := feesCollector.GetCoins().AmountOf(sdk.DefaultStakeDenom)
+	// check for zero fees
+	if feesCollected.IsZero() {
+		return
 	}
-	return total
-}
-
-// store functions used to keep track of a validator award
-func (k Keeper) setValidatorAward(ctx sdk.Ctx, amount sdk.Int, address sdk.Address) {
-	store := ctx.KVStore(k.storeKey)
-	key := types.KeyForValidatorAward(address)
-	val := amino.MustMarshalBinaryBare(amount)
-	store.Set(key, val)
-}
-
-func (k Keeper) getValidatorAward(ctx sdk.Ctx, address sdk.Address) (coins sdk.Int, found bool) {
-	store := ctx.KVStore(k.storeKey)
-	value := store.Get(types.KeyForValidatorAward(address))
-	if value == nil {
-		return sdk.ZeroInt(), false
+	// get the dao and proposer % ex DAO .1 or 10% Proposer .01 or 1%
+	daoAllocation := sdk.NewDec(k.DAOAllocation(ctx))
+	proposerAllocation := sdk.NewDec(k.ProposerAllocation(ctx))
+	totalAllocationWithoutNodeCut := daoAllocation.Add(proposerAllocation)
```

Comment:
Would be less confusing to call this variable `daoAndProposerAllocation` or `totalAllocation`

Reply:
Will fix